src/components/App.tsx

```diff
@@ -291,70 +291,75 @@ export class App extends React.Component<any, AppState> {
               // Perform reconciliation - in collaboration, if we encounter
               // elements with more staler versions than ours, ignore them
               // and keep ours.
-              if (elements == null || elements.length === 0) {
-                elements = restoredState.elements;
+              if (
+                scene.getAllElements() == null ||
+                scene.getAllElements().length === 0
+              ) {
+                this.replaceElements(restoredState.elements);
```

Comment:
What happens if we remove this condition, would it give the same result? If yes, we probably should. This is a very unlikely scenario and makes the code harder to read.

Reply:
can probably be removed, but i didn't add this part. i would prefer to hold off on adding that in this diff since it's a nontrivial change if that's alright

Reply:
All good, ship it without